### PR TITLE
Feat/15669197 datepicker

### PR DESCRIPTION
**Pull request by joanna-pagepro**
I've created a separate PR for Datepicker to merge it before https://github.com/JoinColony/colonyCDapp/pull/1914 and not to block other tasks that require this component.

To check the component, please use Storybook.

This PR includes:

- `Datepicker` component with requested changes from #1914 
- tailwind package update
- `tailwind.config.js` file now includes `base`, `utilities`, and `components` from `main.global.css` to make `Storybook` work and to avoid errors related with global classes (https://github.com/tailwindlabs/tailwindcss/issues/5989#issuecomment-962048436)

### CODE REVIEW

File: src/components/v5/common/Fields/Datepicker/partials/DatepickerCustomRangeInput/DatepickerCustomRangeInput.tsx
Comment (rdig) [inline src/components/v5/common/Fields/Datepicker/partials/DatepickerCustomRangeInput/DatepickerCustomRangeInput.tsx:32]: Can't really find a spec for this, so maybe @melyndav can chime in, but should the inputs allow users to change the values manually, as to me they looks more for display purposes only 

Screenshot from 2024-02-16 18-56-56
Screenshot from 2024-02-16 18-57-03

This is especially relevant since the date format is quite specific, making it nearly impossible for the layman user to actually write this by hand

So what I'm saying is that we still keep the inputs, we just disable them

File: src/components/v5/common/Fields/Datepicker/Datepicker.tsx
Comment (rdig) [inline src/components/v5/common/Fields/Datepicker/Datepicker.tsx:69]: The props passed to these callbacks are _by designed_ made to be changed every single time the specific function is called, so memorizing them makes no sense in this context.

They should just be simple functions, as they don't require all that overhead.

File: src/components/v5/common/Fields/Datepicker/Datepicker.tsx
Comment (rdig) [inline src/components/v5/common/Fields/Datepicker/Datepicker.tsx:121]: I'm really not a fan of mixing global classes and tailwind classes

File: src/components/v5/common/Fields/Datepicker/Datepicker.tsx
Comment (rdig) [inline src/components/v5/common/Fields/Datepicker/Datepicker.tsx:150]: Please rename this variable to variable to `event` and add the proper synthetic event type _(I believe it's `MouseEvent`)_

File: src/components/v5/common/Fields/Datepicker/Datepicker.tsx
Comment (rdig) [inline src/components/v5/common/Fields/Datepicker/Datepicker.tsx:170]: Same here

File: src/components/v5/common/Fields/Datepicker/Datepicker.tsx
Comment (rdig) [inline src/components/v5/common/Fields/Datepicker/Datepicker.tsx:165]: If `selectsRange` is `false`, then `onChangeArg` will not be a tuple, so why are we accessing it's first index value in that case ?

Screenshot from 2024-02-16 19-13-42

File: src/components/v5/common/Fields/Datepicker/partials/DatepickerCustomHeader/DatepickerCustomHeader.tsx
Comment (rdig) [inline src/components/v5/common/Fields/Datepicker/partials/DatepickerCustomHeader/DatepickerCustomHeader.tsx:22]: The min and max year values should be configurable via either props or constants _(same as `DEFAULT_DATE_FORMAT`)_ with a fallback to the values you provided above

File: src/components/v5/common/Fields/Datepicker/partials/DatepickerCustomHeader/DatepickerCustomHeader.tsx
Comment (rdig) [inline src/components/v5/common/Fields/Datepicker/partials/DatepickerCustomHeader/DatepickerCustomHeader.tsx:63]: This should actually use `useLayoutEffect`, as even though most of the time `useEffect` is preferable, in this case it's actually opposite as the changes you are making inside it are observable

File: src/components/v5/common/Fields/Datepicker/partials/DatepickerCustomHeader/DatepickerCustomHeader.tsx
Comment (rdig) [inline src/components/v5/common/Fields/Datepicker/partials/DatepickerCustomHeader/DatepickerCustomHeader.tsx:98]: To some degree this should be tied into `i18n` as internationalization and languages should also change the numbers, time and date formats to they're local regional equivalent.

But I guess that's a larger discussion, so might want to hold off for now.

File: src/i18n/en.json
Comment (rdig) [inline src/i18n/en.json:1577]: This should be placed next to the other `button` message descriptors

**Review by rdig — CHANGES_REQUESTED** [review]
Overall it's mostly there, however it still needs some changes before it can be merged. 

But on the bright side, a lot of good work went in, so nice job! :tada: 

Please be mindful of the comments as well as the Discord discussion about splitting this up: https://discord.com/channels/562263648173555742/998629903740182580/1208063419681603696

Better yet, get in touch with @bassgeta and figure out the proper way to do this.

File: package.json
Comment (joanna-pagepro) [reply package.json:146]: I don't see anything that can break the UI. The changes presented in this release include some utilities, that were requested by some of you in other PRs, like using `min-h-*` className with size variable. In many cases, tailwind's documentation is updated to reflect the changes from this update.
I won't upgrade anything else without asking you first. Sorry about this one.

File: tailwind.config.js
Comment (joanna-pagepro) [reply tailwind.config.js:153]: Sometimes there was an error in the console, when you were running the project, that some class does not exists and to add it using the `@layer` directive and here's the explanation from the [issue that I've pasted in line 151](https://github.com/tailwindlabs/tailwindcss/issues/5989#issuecomment-962048436):
```
Again, what is happening is that your build tool is effectively running all of these processes in parallel:

tailwindcss -i global.css
tailwindcss -i module-one.module.css
tailwindcss -i module-two.module.css
tailwindcss -i module-three.module.css
When you realize this it is a bit less surprising that code in module-one.module.css can't "see" code in global.css. But if they share a config file and the code is in the config file, you are all good.
```
By moving everything that was using this `@layer` directive from `main.global.css` to `tailwind.config.js` I have resolved the problem described above. Because of this issue, I wasn't able to run Storybook and it would be hard to test Datepicker component, because it's not used anywhere in the app right now (in this PR).

File: src/components/v5/common/Fields/Datepicker/Datepicker.tsx
Comment (joanna-pagepro) [reply src/components/v5/common/Fields/Datepicker/Datepicker.tsx:121]: I'm using this class with CSS Modules. With tailwind classes, it would be pretty hard to modify some elements based on other classes added to the Datepicker by default.

**Comment by joanna-pagepro** [comment]
There are two separate components now.

@rdig I've changed the inputs that are added to datepicker header to be read-only - I think that you're right and that they shouldn't be editable.

**Review by bassgeta — APPROVED** [review]
looks good now, thanks for splitting them up!

**Review by rdig — APPROVED** [review]
Looks good, and I like the way you split up the datepicker / range datepicker components as it makes it more readable.

Good to go.

Screenshot from 2024-02-19 21-37-00

**Pull request merged**
